File: perfbench-code/swicket-jpa/src/main/scala/swicketjpa/wicket/TemplatePage.scala
package swicketjpa.wicket

import org.apache.wicket.markup.html.WebPage
import org.apache.wicket.markup.html.basic.Label
import org.apache.wicket.markup.html.link.BookmarkablePageLink
import org.apache.wicket.model.PropertyModel
import org.slf4j.LoggerFactory
import swicketjpa.entity.Booking

object TemplatePage {
  val logger = LoggerFactory.getLogger(classOf[TemplatePage])
}

class TemplatePage extends WebPage {

  def logger = TemplatePage.logger

  add(new Label("userName", new PropertyModel(this, "session.user.name")))
  add(new BookmarkablePageLink("search", classOf[MainPage]))
  add(new BookmarkablePageLink("settings", classOf[PasswordPage]))
  add(new BookmarkablePageLink("logout", classOf[LogoutPage]))

  def getEntityManager() = getRequestCycle().asInstanceOf[JpaRequestCycle].getEntityManager()

  def endConversation() = getRequestCycle().asInstanceOf[JpaRequestCycle].endConversation()

  def getBookingSession() = getSession().asInstanceOf[BookingSession]

  def loadBookings() = {
    val query = getEntityManager().createQuery("select b from Booking b"
      + " where b.user.username = :username order by b.checkinDate")
    query.setParameter("username", getBookingSession().user.username)
    getBookingSession().bookings = query.getResultList().asInstanceOf[java.util.List[Booking]]
  }

}

File: perfbench-code/wicket6-jpa/src/main/java/wicketjpa/wicket/TemplatePage.java
package wicketjpa.wicket;

import javax.persistence.EntityManager;
import javax.persistence.Query;
import org.apache.wicket.markup.html.WebPage;
import org.apache.wicket.markup.html.basic.Label;
import org.apache.wicket.markup.html.link.BookmarkablePageLink;
import org.apache.wicket.model.PropertyModel;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

public class TemplatePage extends WebPage {

    protected static final Logger logger = LoggerFactory.getLogger(TemplatePage.class);        

    public TemplatePage() {
        add(new Label("userName", new PropertyModel(this, "session.user.name")));
        add(new BookmarkablePageLink<Void>("search", MainPage.class));
        add(new BookmarkablePageLink<Void>("settings", PasswordPage.class));
        add(new BookmarkablePageLink<Void>("logout", LogoutPage.class));
    }

    protected EntityManager getEntityManager() {
        return JpaRequestCycleListener.getEntityManager();
    }

    protected void endConversation() {
        JpaRequestCycleListener.endConversation();
    }

    protected BookingSession getBookingSession() {
        return (BookingSession) getSession();
    }

    protected void loadBookings() {
        Query query = getEntityManager().createQuery("select b from Booking b"
                + " where b.user.username = :username order by b.checkinDate");
        query.setParameter("username", getBookingSession().getUser().getUsername());
        getBookingSession().setBookings(query.getResultList());
    }    
}
